modular_zubbers/code/modules/hydroponics/advancedplantbag.dm

```diff
@@ -0,0 +1,47 @@
+/datum/storage/adv_plant_bag    //The storage type for the bag below
+
+/datum/storage/adv_plant_bag/remove_single(mob/removing, obj/item/thing, atom/newLoc, silent)
+    var/obj/item/resolve_location = real_location?.resolve()
+    if(!resolve_location)
+        return
+
+    resolve_location.visible_message(span_notice("[removing] reaches inside \the [resolve_location]."),
+        span_notice("You reach into \the [resolve_location] to grab something."))
+    if(!do_after(removing, 1.5 SECONDS, resolve_location))    //A short period to grab an item to avoid mid-combat abuse, taken from bluespace trash bags.
+        return
```

Comment:
Does not seem to be a weakref anymore?
```suggestion
/datum/storage/adv_plant_bag/remove_single(mob/removing, obj/item/thing, atom/newLoc, silent)
    real_location.visible_message(span_notice("[removing] reaches inside \the [resolve_location]."),
        span_notice("You reach into \the [resolve_location] to grab something."))
    if(!do_after(removing, 1.5 SECONDS, resolve_location))    //A short period to grab an item to avoid mid-combat abuse, taken from bluespace trash bags.
        return
```